File: Tut3_ROS_Service/src/add_two_int/scripts/add_two_ints_server.py
#!/usr/bin/env python3

import rospy
from rospy_tutorials.srv import AddTwoInts

def handle_add_two_ints(req):
    result = req.a +req.b
    rospy.loginfo('Sum of '+ str(req.a) +' and '+ str(req.b)+ ' is '+ str(result))
    return result

if __name__ == '__main__':

    #select service name
    nodeName = "add_two_ints_server"

    #select service name
    serviceName = "/add_two_ints"

    #Init Node with its name
    rospy.init_node(nodeName)

    #Send a Log msg in terminal
    rospy.loginfo(nodeName+" has been created")

    #configure service with msg and callb-back
    service = rospy.Service(serviceName,AddTwoInts, handle_add_two_ints)

    rospy.spin()

File: Tut3_ROS_Service/src/add_two_int/src/add_two_ints_server.cpp
#include <ros/ros.h>
#include <string>
#include <rospy_tutorials/AddTwoInts.h>

bool handl_add_two_ints(rospy_tutorials::AddTwoInts::Request &req, rospy_tutorials::AddTwoInts::Response &resp)
{
    int result(req.a + req.b);
    const std::string sum("The Sum of"+ std::to_string(req.a) + " and " + std::to_string(req.a) + " ---> "
                          + std::to_string(result));
    ROS_INFO(sum.c_str());
    resp.sum = result;
    return true;
}

int main(int argc, char **argv)
{
    //select Ros node name
    const std::string nodeName("add_two_ints_server_roscpp");

    //select Ros topic name
    const std::string serviceName("/add_two_ints_server");

    //Init Node with its name
    ros::init(argc, argv, nodeName);

    //create node handle
    ros::NodeHandle nh;

    //Configure node node type(publisher) , the topic Name, and queue size
    ros::ServiceServer server(nh.advertiseService(serviceName, handl_add_two_ints));

    //Select the rate of node execution
    ros::spin();
}